examples/agent-sandbox-rl/agent_sandbox_rl/resources.py

```
@@ -30,9 +30,48 @@
+      return func(*args, **kwargs)
+    except client.ApiException as e:
+      if e.status == 429 and attempt < max_retries - 1:
+        retry_after = 1.0
+        if e.headers:
+          raw_retry = e.headers.get("Retry-After") or e.headers.get("retry-after")
+          if raw_retry:
+            try:
+              retry_after = float(raw_retry)
+            except (ValueError, TypeError):
+              pass
+        sleep_time = max(retry_after, backoff) + random.uniform(0.1, 0.5)
```

Ruff (0.16.0)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)